addons/web_unsplash/static/src/frontend/unsplash_beacon.js

```diff
@@ -1,29 +1,32 @@
-import publicWidget from "@web/legacy/js/public/public_widget";
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
 import { rpc } from "@web/core/network/rpc";
 
-publicWidget.registry.UnsplashBeacon = publicWidget.Widget.extend({
-    // /!\ To adapt the day the beacon makes sense for backend customizations
-    selector: '#wrapwrap',
+export class UnsplashBeacon extends Interaction {
+    static selector = "#wrapwrap";
+
+    async willStart() {
+        const unsplashImageEls = this.el.querySelectorAll('img[src*="/unsplash/"]');
+        const unsplashImageIds = [];
+        for (const unsplashImageEl of unsplashImageEls) {
+            // extract the image id from URL (`http://www.domain.com:1234/unsplash/xYdf5feoI/lion.jpg` -> `xYdf5feoI`)
+            unsplashImageIds.add(unsplashImageEl.src.split('/unsplash/')[1].split('/')[0]);
+        }
 
-    /**
-     * @override
-     */
-    start: function () {
-        var unsplashImages = Array.from(this.$('img[src*="/unsplash/"]')).map((img) => {
-            // get image id from URL (`http://www.domain.com:1234/unsplash/xYdf5feoI/lion.jpg` -> `xYdf5feoI`)
-            return img.src.split('/unsplash/')[1].split('/')[0];
-        });
-        if (unsplashImages.length) {
-            rpc('/web_unsplash/get_app_id').then(function (appID) {
-                if (!appID) {
-                    return;
-                }
-                $.get('https://views.unsplash.com/v', {
-                    'photo_id': unsplashImages.join(','),
+        if (unsplashImageIds.length) {
+            const appID = await this.waitFor(rpc('/web_unsplash/get_app_id'));
+
+            if (appID) {
+                fetch('https://views.unsplash.com/v', {
+                    'photo_id': unsplashImageIds.join(','),
                     'app_id': appID,
-                });
-            });
+                })
```

Comment:
`photo_id` and `app_id` are actually query parameters, so this syntax won't work. Put them in a `new URLSearchParams()` ;)